### ISSUE TITLE
NumPy 2.0 AttributeError

### ISSUE THREAD

**Issue by RDWimmers** [mbakker7/ttim#65]
set upper limit for NumPy dependency or update code base to support NumPy 2.0

```
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/ttim/__init__.py:13: in <module>
    from .linedoublet import LeakyLineDoublet, LeakyLineDoubletString
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/ttim/linedoublet.py:6: in <module>
    from . import besselnumba
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/ttim/besselnumba.py:25: in <module>
    nrange = np.arange(21, dtype=np.float_)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/numpy/__init__.py:[39]
E   AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.
```

**Comment by mbakker7** [comment]
Thanks for the report. We are working on Numpy 2.0 compatibility. Until that update, stick with Numpy < 2.0

**Comment by mbakker7** [comment]
Fixed with release 0.6.7
